src/testers.jl

```diff
@@ -112,61 +112,55 @@ at input point `z` to confirm that there are correct `frule` and `rrule`s provid
 
 `fkwargs` are passed to `f` as keyword arguments.
 All keyword arguments except for `fdm` and `fkwargs` are passed to `isapprox`.
+
+To use this tester for a scalar type `MyNumber <: AbstractNumber`,
+`FiniteDifferences.to_vec(::MyNumber)` must be implemented.
 """
 function test_scalar(f, z; rtol=1e-9, atol=1e-9, fdm=_fdm, fkwargs=NamedTuple(), kwargs...)
```

Comment:
It would simplify the `frule` test because we wouldn't need the basis, but if the output is non-real we still need the basis on the output for the `rrule` test. Adding a separate method would require us to maintain that code in two places.